File: Java/triforce.java
/**
 * ================================================================================
 *                       Date of development: July 8th, 2023
 *                         Last modified: July 9th, 2023
 * ================================================================================
 *
 * ================================================================================
 *               This program draws a Triforce shape using recursion.
 *
 * It starts by drawing the largest equilateral triangle and recursively calls itself
 *        to draw smaller Triforce shapes within each larger triangle.
 *
 * The size of the Triforce shape is controlled by the 'size' parameter passed to
 *     the 'drawTriforce' method. The program positions the Triforce shape near the
 *                          bottom center of the screen.
 * ================================================================================
 * ================================================================================
 */

import java.awt.Color;
import java.awt.Graphics;
import javax.swing.JFrame;
import javax.swing.JPanel;

public class triforce extends JPanel {
    private static final int WIDTH = 600;
    private static final int HEIGHT = 600;
    
    public void paintComponent(Graphics g) {
        super.paintComponent(g);
        
        // Set background color
        setBackground(Color.BLACK);
        
        // Set drawing color
        g.setColor(Color.WHITE);
        
        // Draw the triforce shape
        drawtriforce(g, WIDTH / 2, HEIGHT, HEIGHT);
    }
    
    private void drawTriangle(Graphics g, int x, int y, int size) {
        int[] xPoints = {x, x - size / 2, x + size / 2};
        int[] yPoints = {y, y - size, y - size};
        g.fillPolygon(xPoints, yPoints, 3);
    }
    
    private void drawtriforce(Graphics g, int x, int y, int size) {
        if (size < 10) {
            return;
        }
        
        // Draw the current triangle
        drawTriangle(g, x, y, size); 
        
        // Recursive calls to draw the smaller triforce shapes
        drawtriforce(g, x, y - size, size / 2);
        
        drawtriforce(g, x - size / 2, y, size / 2);
        
        drawtriforce(g, x + size / 2, y, size / 2);
    }
    
    public static void main(String[] args) {
        JFrame frame = new JFrame("triforce");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setSize(WIDTH, HEIGHT);
        
        triforce panel = new triforce();
        frame.add(panel);
        
        frame.setVisible(true);
    }
}

File: Python/triforce.py
#################################################################################
##                                Can You Even Triforce?                       ##
##                         Date of development: July 8th, 2023                 ##
##                          Last modified: July 9th, 2023                      ##
##                               Designed by: Ryan Hatch                       ##
#################################################################################

################################################################################# 
##               This code uses recursion to draw a Triforce shape.            ##
##                                                                             ##
## It starts by drawing the largest equilateral triangle, then recursively     ##
## calls itself to draw smaller Triforce shapes within each larger triangle.   ##
##                                                                             ##
##               The length parameter controls the size of the Triforce shape. ##
## The code positions the turtle at (-300, -300) to ensure the entire shape is ##
##                             visible on the screen.                          ##
#################################################################################

import turtle

def draw_triangle(t, length):
    # Draw an equilateral triangle
    for _ in range(3):
        t.forward(length)
        t.left(120)

def draw_triforce(t, length):
    
    # Base case for recursion
    if length < 10:  
        return
    
    # Draw the current triangle
    draw_triangle(t, length)
    
    # Recursive calls to draw the smaller Triforce shapes
    draw_triforce(t, length/2)
    
    t.forward(length/2)
    draw_triforce(t, length/2)
    t.backward(length/2)
    
    t.left(60)
    t.forward(length/2)
    t.right(60)
    draw_triforce(t, length/2)
    t.left(60)
    t.backward(length/2)
    t.right(60)

# Create a turtle object
t = turtle.Turtle()
t.speed("fast")

# Position the turtle to start drawing the Triforce
t.penup()
t.goto(-300, -300)
t.pendown()

# Draw the Triforce shape
draw_triforce(t, 600)

# Hide the turtle
t.hideturtle()

# Keep the screen open until closed manually
turtle.done()